include/revng/Pipeline/LLVMKind.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <iterator>
+
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Analysis/ScalarEvolutionExpressions.h"
+#include "llvm/IR/Function.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+
+#include "revng/Pipeline/ContainerEnumerator.h"
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Kind.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Target.h"
+
+namespace Pipeline {
+
+/// Kind that must be extended to be able to specify how a to deduce
+/// the target associated to a particular llvm global function.
+///
+/// When extented two function must be provided, symbolToTarget must return
+/// the target associated to a function if there is any
+///
+/// compactTargets must collapse the targets into the * target if they are all
+/// presents, do no thing otherwise.
+template<typename LLVMContainer>
+class LLVMKindBase : public KindForContainer<LLVMContainer> {
```

Comment:
This `Kind` seems to be opinionated on being associated to a `llvm::Function`. If it does not make sense to use it for anything else, I'd call it `LLVMFunctionKindBase`.

Reply:
renamed it LLVMGlobalKindBase since it handles all top level objects in the module